extensions/notebook/src/book/githubRemoteBook.ts

```diff
@@ -0,0 +1,88 @@
+import * as request from 'request';
+import * as fs from 'fs-extra';
+import * as loc from '../common/localizedConstants';
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as zip from 'adm-zip';
+import * as tar from 'tar';
+import * as utils from '../common/utils';
+import { RemoteBook } from './remoteBook';
+import { IAsset } from './remoteBookController';
+
+export class GitHubRemoteBook extends RemoteBook {
+	constructor(public remotePath: URL, protected _asset: IAsset) {
+		super(remotePath, _asset);
+	}
+
+	public async createLocalCopy(): Promise<void> {
+		this.outputChannel.show(true);
+		this.outputChannel.appendLine(loc.msgRemoteBookDownloadProgress);
+		this.setLocalPath();
+		this.createDirectory();
+
+		return new Promise((resolve, reject) => {
+			let options = {
+				headers: {
+					'User-Agent': 'request'
+				}
+			};
+			let downloadRequest = request.get(this._asset.browserDownloadUrl.href, options)
+				.on('error', (downloadError) => {
+					this.outputChannel.appendLine(loc.msgRemoteBookDownloadError);
+					reject(downloadError);
+				})
+				.on('response', (response) => {
+					if (response.statusCode !== 200) {
+						this.outputChannel.appendLine(loc.msgRemoteBookDownloadError);
+						return reject(response.statusMessage);
+					}
+				});
+			let remoteBookFullPath = new URL(this._localPath.href.concat('.zip'));
+			downloadRequest.pipe(fs.createWriteStream(remoteBookFullPath.href))
+				.on('close', async () => {
+					resolve(this.extractFiles(remoteBookFullPath));
+				})
+				.on('error', (downloadError) => {
+					this.outputChannel.appendLine(loc.msgRemoteBookDownloadError);
+					reject(downloadError);
+					downloadRequest.abort();
+				});
+		});
+	}
+	public async createDirectory(): Promise<void> {
+		let fileName = this._asset.book.concat('-').concat(this._asset.version).concat('-').concat(this._asset.language);
+		this._localPath = new URL(path.join(this._localPath.href, fileName));
+
+		try {
+			let exists = await fs.pathExists(this._localPath.href);
+			if (exists) {
+				await fs.remove(this._localPath.href);
+			}
+			await fs.promises.mkdir(this._localPath.href);
+		} catch (error) {
+			this.outputChannel.appendLine(error);
+		}
+	}
+	public async extractFiles(remoteBookFullPath: URL): Promise<void> {
+		try {
+			if (utils.getOSPlatform() === utils.Platform.Windows || utils.getOSPlatform() === utils.Platform.Mac) {
+				let zippedFile = new zip(remoteBookFullPath.href);
+				zippedFile.extractAllTo(this._localPath.href);
+			} else {
+				tar.extract({ file: remoteBookFullPath.href, cwd: this._localPath.href }).catch(err => {
+					this.outputChannel.appendLine(loc.msgRemoteBookUnpackingError);
+				});
+			}
+			await fs.promises.unlink(remoteBookFullPath.href);
+			this.outputChannel.appendLine(loc.msgRemoteBookDownloadComplete);
+			vscode.commands.executeCommand('notebook.command.openNotebookFolder', this._localPath.href);
```

Comment:
>openNotebookFolder [](start = 52, length = 18)

Why aren't you using openBook command instead if it's always going to be a JupyterBook? I know we have had a discussion around this, totally forgot to why you had to use openNotebookFolder.

Reply:
You're right. For the previous implementation, I downloaded the whole repo, instead of a selected asset, that's why it made sense to use the openNotebookFolder command. I'll change it to use the openBook, instead. Thanks 😄!

Reply:
After changing it back to openBook, I remembered why we're using the openNotebookFolder command, instead.
OpenBook opens file explorer when given the bookPath, and since remoteBookDialog already works like a book selector it seems very repetitive to open a dialog and then open file explorer, too.